std/security/ed-signer.go

```diff
@@ -0,0 +1,51 @@
+package security
+
+import (
+	"crypto/ed25519"
+	"time"
+
+	enc "github.com/named-data/ndnd/std/encoding"
+	basic_engine "github.com/named-data/ndnd/std/engine/basic"
+	"github.com/named-data/ndnd/std/ndn"
+)
+
+// edSigner is a signer that uses Ed25519 key to sign packets.
+type edSigner struct {
```

Comment:
also nitpicking but much less important than the previous comment: EdDSA can be instantiated with different curves and hash algorithms. Ed25519 is just one instance (arguably the most common one) but the name `edSigner` could potentially indicate that this is for EdDSA generically. Thankfully, the comment is clear that it's for Ed25519, so this is not too important at the end of the day. It could become awkward when/if we introduce the Ed448 signature type.